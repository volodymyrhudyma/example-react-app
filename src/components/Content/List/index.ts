import { connect } from 'react-redux';
import { IAppState } from 'src/types';
import Component from './List';
import { fetchArticles } from 'src/actions/article';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  layout: state.layout.layout,
  fetching: state.article.articles.fetching,
  error: state.article.articles.error,
  data: state.article.articles.data,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    fetchArticles: () => dispatch(fetchArticles()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
