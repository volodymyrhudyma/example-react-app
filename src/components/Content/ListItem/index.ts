import { connect } from 'react-redux';
import { IAppState, IArticle } from 'src/types';
import Component from './ListItem';
import { readArticle, deleteArticle } from 'src/actions/article';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  layout: state.layout.layout,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    readArticle: (article: IArticle) => dispatch(readArticle(article)),
    deleteArticle: (id: number) => dispatch(deleteArticle(id)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
