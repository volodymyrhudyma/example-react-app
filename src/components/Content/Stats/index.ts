import Component from './Stats';
import { connect } from 'react-redux';
import { IAppState, IArticle } from 'src/types';
import { readArticle } from 'src/actions/article';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  articles: state.article.articles.data,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    readArticle: (article: IArticle) => dispatch(readArticle(article)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
