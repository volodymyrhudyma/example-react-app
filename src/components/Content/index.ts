import { connect } from 'react-redux';
import { setLayout } from 'src/actions/layout';
import { IAppState } from 'src/types';
import Component from './Content';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  layout: state.layout.layout,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    setLayout: (layout: string) => dispatch(setLayout(layout)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
