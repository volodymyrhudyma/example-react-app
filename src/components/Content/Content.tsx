import React, { Component } from 'react';
import { Wrapper, ContentInner, TitleWrapper } from './styled/Content.styled';
import Container from 'src/ui-kit/Container';
import Title from 'src/ui-kit/Title';
import Stats from './Stats';
import List from './List';
import Tabs from 'src/ui-kit/Tabs';
import icons from 'src/constants/icons';
import { TABLE, CARD } from 'src/constants/layout';
import { WithTranslation } from 'react-i18next';

interface IContentProps extends WithTranslation {
  layout: string;
  setLayout: (layout: string) => void;
}

class Content extends Component<IContentProps> {
  onCardsViewClick = () => {
    const { setLayout } = this.props;
    setLayout(CARD);
  };
  onTableViewClick = () => {
    const { setLayout } = this.props;
    setLayout(TABLE);
  };
  render() {
    const { t, layout } = this.props;
    const activeTabIndex = layout === TABLE ? 1 : 0;
    return (
      <Wrapper>
        <Container>
          <ContentInner>
            <TitleWrapper>
              <Title type="h1">{t('articles.title')}</Title>
              <Tabs
                activeTabIndex={activeTabIndex}
                items={[
                  {
                    icon: {
                      default: icons.card,
                      active: icons.cardActive,
                    },
                    label: t('layout.cards'),
                    onClick: this.onCardsViewClick,
                  },
                  {
                    icon: {
                      default: icons.table,
                      active: icons.tableActive,
                    },
                    label: t('layout.table'),
                    onClick: this.onTableViewClick,
                  },
                ]}
              />
            </TitleWrapper>
            <Stats />
            <List />
          </ContentInner>
        </Container>
      </Wrapper>
    );
  }
}

export default Content;
