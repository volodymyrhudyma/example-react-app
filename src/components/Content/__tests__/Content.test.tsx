import { shallow } from 'enzyme';
import React from 'react';
import Content from '../Content';
import i18n from 'i18next';

describe('Content with Enzyme and Jest', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Content
        t={jest.fn()}
        i18n={i18n}
        tReady={false}
        layout="TABLE"
        setLayout={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
