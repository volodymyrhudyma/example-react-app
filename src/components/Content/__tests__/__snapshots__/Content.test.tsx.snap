// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content with Enzyme and Jest should render correctly 1`] = `
<styled.div>
  <Container>
    <styled.div>
      <styled.div>
        <Title
          type="h1"
        />
        <Tabs
          activeTabIndex={1}
          items={
            Array [
              Object {
                "icon": Object {
                  "active": "/icons/card-active.svg",
                  "default": "/icons/card.svg",
                },
                "label": undefined,
                "onClick": [Function],
              },
              Object {
                "icon": Object {
                  "active": "/icons/table-active.svg",
                  "default": "/icons/table.svg",
                },
                "label": undefined,
                "onClick": [Function],
              },
            ]
          }
        />
      </styled.div>
      <Connect(withI18nextTranslation(Stats)) />
      <Connect(withI18nextTranslation(List)) />
    </styled.div>
  </Container>
</styled.div>
`;
